packages/flutter/lib/src/foundation/_platform_web.dart

```diff
@@ -4,52 +4,12 @@
 
 import 'dart:ui' as ui;
 
-import 'package:js/js.dart';
+import '../dom.dart';
 
 import 'platform.dart' as platform;
 
 export 'platform.dart' show TargetPlatform;
 
-/// [DomWindow] interop object.
-@JS()
-@staticInterop
-class DomWindow {}
-
-/// [DomWindow] required extension.
-extension DomWindowExtension on DomWindow {
-  /// Returns a [DomMediaQueryList] of the media that matches [query].
-  external DomMediaQueryList matchMedia(String? query);
-
-  /// Returns the [DomNavigator] associated with this window.
-  external DomNavigator get navigator;
-}
-
-/// The underyling window.
-@JS('window')
-external DomWindow get domWindow;
-
-/// [DomMediaQueryList] interop object.
-@JS()
-@staticInterop
-class DomMediaQueryList {}
```

Comment:
I thought we wanted to have some duplication here, but I prefer this to be centralized in dom.dart.

Reply:
Agreed!